uiowa/uids
# Storybook 8 migration

# How to test

- `yarn install` 
- `yarn storybook`
- Compare against http://uids.brand.uiowa.edu/latest
- Click through each component story, test controls, verify that HTML is still available. 
- Review code changes and try to identify anything that could still be removed or anything that has been removed that we still might need. 


## Test build process:

-    `yarn production`
-    `npx http-server ./www`
-     Go to the Introduction page and confirm that CSS files are listed using the version number from package.json.
-     Go to the Brand Bar docs page and confirm that the CSS files are listed using the version number from package.json.



Deprecated Items changed:
- @storybook/preset-scss has been [deprecated](https://github.com/storybookjs/presets/tree/master/packages/preset-scss).  Moved to `storybook-addon-sass-postcss` but it looks like there are other options too. 
- Deprecation Warning: The legacy JS API is deprecated and will be removed in Dart Sass 2.0.0.  More info: https://sass-lang.com/d/legacy-js-api
- Deprecation Warning: Sass's behavior for declarations that appear after nested
rules will be changing to match the behavior specified by CSS in an upcoming
version. To keep the existing behavior, move the declaration above the nested
rule. To opt into the new behavior, wrap the declaration in `& {}`. More info: https://sass-lang.com/d/mixed-decls
- Deprecation Warning: lighten() is deprecated. More info: https://sass-lang.com/d/color-functions
  - Switched to `@use "sass:color";`

Deprecated Items todo in another issue:
- sass import rules are deprecated and will be removed in Dart Sass 3.0.0. More info and automated migrator: https://sass-lang.com/d/import
